compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParserErrorHandler.java

```diff
@@ -1857,10 +1873,18 @@ protected ParserRuleContext getNextRule(ParserRuleContext currentCtx, int nextLo
             case TABLE_CONSTRUCTOR:
                 return ParserRuleContext.TABLE_KEYWORD;
             case TABLE_KEYWORD:
+                parentCtx = getParentContext();
+                if (parentCtx == ParserRuleContext.TABLE_TYPE_DESCRIPTOR) {
+                    return ParserRuleContext.LT;
```

Comment:
An alternative way is to give an alias to `<` as use it here. For example say `row-type-param-start` as the next context.

Then inside the getNextRule() method, for the `row-type-param-start`, we can give the `FOO` (which i mentioned earlier) as the next-rule.